added ability to directly attach a stream

antoniosou opened the pull request:
This is helpful because you won't have to create an Attachment object for streams and figure out the contentType and knownLength. If you wanted to attach multiple files from the internet for example, you could very easily do this (using request module) by setting attachment: [request(url1), request(url2), request(url3), ...]

bojand commented:
Thanks for the PR. This would be useful. Can you please add a test for this functionality in the file `test/mailgun.test.js` that fetches and attaches a simple file, perhaps the [Google logo](https://www.google.ca/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png). Thanks.

antoniosou commented:
Added a test and updated the README file. I followed what you did for the other attachment tests, however, I'm not sure those tests can verify that a file has been attached, I think they just verify that the message has been sent.

bojand commented:
Thanks. I'll test it out.
